ten-simple-rules-dockerfiles.Rmd

```diff
@@ -355,7 +379,7 @@ It is common to install all software as the default user `root`.
 In fact it is _required_ for system dependencies.
 You can switch the user running subsequent `RUN`, `CMD` and `ENTRYPOINT` instructions with the `USER` instruction, but you have to make sure yourself that the user exists within the image.
 This is useful if a software aborts installations as the `root` user, or to avoid permission problems when using the container (see&nbsp;\ruleref{rule:mount}).
-Be sure to document a non-default user in detail and to examine the used users in base images if you encounter errors on user permissions.
+Be sure to document a non-default user in detail and to examine the used users in base images if you encounter errors on user permissions.  [SJE: okay, this para confused me -- just tell me what to do!  should I not bother creating a user, and run as root, or should I create a user?  rstudio complictes matters here I think, don't they create a "rstudio" username?]
```

Comment:
For most scientific containers that run on HPC, the conversion to Singularity makes the user directive useless. For docker containers that run on the host, using a user with specific permissions usually makes it harder for the person to get it working. If you are going to use Docker and take the security implications with that, I think the path to least resistance is using standard root, but not installing things in a way that would break if run in a user space (e.g., no executables in root's home, etc.)

Reply:
Yes, Rocker images have an `rstudio` user. Not having a non-root user can also be pretty annoying if you mount files into containers and cannot delete them (without sudo-ing) on the host afterwards...

I like the suggestion ("make sure root works"), and we should update the text accordingly. At the same time we should concisely mentioning the potential complications of users, because the reader might come across them via used base images.